pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -149,34 +268,431 @@ func (c *Check) instanceConfigure(data integration.Data) error {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+				}
+			}
+		}
+	}
+	for _, key := range []string{"device_exclude", "device_blacklist", "excluded_disks"} {
+		if deviceExclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+				}
+			}
+		}
+	}
+	excludedDiskRe, found := instanceConfig["excluded_disk_re"]
+	if excludedDiskRe, ok := excludedDiskRe.(string); found && ok {
+		var err error
+		regexp, err := regexp.Compile(excludedDiskRe)
+		if err != nil {
+			return err
+		}
+		c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+	}
 	return nil
 }
 
-func stringSliceContain(slice []string, x string) bool {
-	return slices.Contains(slice, x)
+func (c *Check) configureIncludeDevice(instanceConfig map[interface{}]interface{}) error {
+	for _, key := range []string{"device_include", "device_whitelist"} {
+		if deviceInclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range deviceInclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.includedDevices = append(c.cfg.includedDevices, *regexp)
+				}
+			}
+		}
+	}
+	return nil
 }
 
-func (c *Check) applyDeviceTags(device, mountpoint string, tags []string) []string {
-	// apply device/mountpoint specific tags
-	for re, deviceTags := range c.cfg.deviceTagRe {
-		if re == nil {
+func (c *Check) configureExcludeFileSystem(instanceConfig map[interface{}]interface{}, initConfig map[interface{}]interface{}) error {
+	defaultFileSystems := []interface{}{"iso9660$", "tracefs$"}
+	var fileSystemExclude []interface{}
+	for _, key := range []string{"file_system_global_exclude", "file_system_global_blacklist"} {
+		if val, ok := initConfig[key].([]interface{}); ok {
+			fileSystemExclude = val
+			break // Stop at the first valid key found
+		}
+	}
+	// Use default values if neither key was found
+	if fileSystemExclude == nil {
+		fileSystemExclude = defaultFileSystems
+	}
+	for _, val := range fileSystemExclude {
+		if strVal, ok := val.(string); ok {
+			if re, err := regexp.Compile(strVal); err == nil {
+				c.cfg.excludedFilesystems = append(c.cfg.excludedFilesystems, *re)
+			} else {
+				return err
+			}
+		}
+	}
+	for _, key := range []string{"file_system_exclude", "file_system_blacklist", "excluded_filesystems"} {
+		if fileSystemExclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range fileSystemExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
```

Comment:
The Python versions adds a `$` at the end of the regex for those configurations
https://github.com/DataDog/integrations-core/blob/master/disk/datadog_checks/disk/disk.py#L364